include/wil/cppwinrt_authoring.h

```diff
@@ -0,0 +1,236 @@
+//*********************************************************
+//
+//    Copyright (c) Microsoft. All rights reserved.
+//    This code is licensed under the MIT License.
+//    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF
+//    ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
+//    TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+//    PARTICULAR PURPOSE AND NONINFRINGEMENT.
+//
+//*********************************************************
+
+namespace wil
+{
+#ifndef __WIL_CPPWINRT_AUTHORING_INCLUDED
+#define __WIL_CPPWINRT_AUTHORING_INCLUDED
+    namespace details
+    {
+        template<typename T>
+        struct single_threaded_property_storage
+        {
+            T m_value{};
+            single_threaded_property_storage() = default;
+            single_threaded_property_storage(const T& value) : m_value(value) {}

```

Comment:
```suggestion
            single_threaded_property_storage(T value) : m_value(std::move(value)) {}
```

Force the caller to do the copy and then move that value into the local member. This has a downside if the type is not movable, however that's _probably_ not the case for any type that this will apply to. An alternative would be to provide two constructors: one taking `const T&` and another taking `T&&`.

If you want to go off the crazy deep end, accept `Args&&...` and forward-construct the member.

Reply:
Though it looks like this is only used for scalar types? If so, this suggestion as well as the templated `operator=` below seems like it's probably overkill.

That said, the general idea of the suggestion applies below.